crates/dojo-world/src/contracts/world.rs

```diff
@@ -11,29 +11,13 @@ use super::model::{ModelError, ModelRPCReader};
 #[path = "world_test.rs"]
 pub(crate) mod test;
 
-#[cfg(not(doctest))]
 pub mod abigen {
     pub mod world {
-        use cainome::rs::abigen;
-        abigen!(WorldContract, "crates/dojo-world/src/contracts/abi/world.json");
```

Comment:
could we use something like std::include_str! to embed the abi json so we don't need the dev to install a new cli util? it would be a simpler pipeline: build artifacts outputted by dojo would automatically be updated when abigen is called during compilation. we could add the embed to cainome directly and handle the embedding there so the macro interface is the same

Reply:
Yes that's definitely good to have the binding generated at the compile time. I've tried to push this way for a more robust and simpler pipeline.

The situation I have here is the following:
1. We can't use `include_str!` inside abigen as the abigen receives the not expanded `include_str!` tokens. Eager expansion is postpone on rustc roadmap.
2. We thus can build the dojo-core contracts without using dojo plugin to avoid circular dependencies, and then generate the binding file for cainome as done in the new build.rs of `dojo-world`. 
3. Seems like I can't rename files in the CI.

Reply:
The issue is related to a project that is external to dojo building without the dojo repo context. This causes some files to not be found as the directory structure is not the same.

To follow your comment @tarrencev and to keep CI validating consistency between the bindings and the `dojo-core` contracts, I propose in the last commits a small tool to embed all the logic to build the bindings and a verification that is run in the CI to have our back.

Please let me know your opinion on that, same for @kariy.